File: mcp-server/internal/domains/workflows/resources.go
package workflows

import (
	"github.com/modelcontextprotocol/go-sdk/mcp"

	"github.com/stigmer/stigmer/mcp-server/internal/domains"
)

// Template returns the MCP resource template for workflows.
func Template() *mcp.ResourceTemplate {
	return &mcp.ResourceTemplate{
		URITemplate: "stigmer://workflows/{org}/{slug}",
		Name:        "stigmer_workflow",
		Title:       "Stigmer Workflow",
		Description: "Full definition of a Stigmer workflow, identified by organization and slug.",
		MIMEType:    "application/json",
	}
}

// ResourceHandler returns a handler that reads a workflow resource by parsing
// the org and slug from the request URI.
func ResourceHandler(serverAddress string) mcp.ResourceHandler {
	return domains.NewResourceHandler(Fetch, serverAddress, "workflows")
}

File: mcp-server/src/domains/workflows/resources.ts
// Workflow resource template (stigmer://workflows/{org}/{slug}).
// Go parity: mcp-server/internal/domains/workflows/resources.go.

import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

import type { BackendTarget } from "../client.js";
import { registerResource } from "../resourcehandler.js";
import { fetchWorkflow } from "./fetch.js";

/** Register the workflow resource template; returns the registered resource names. */
export function registerWorkflowResources(server: McpServer, target: BackendTarget): string[] {
  registerResource(server, target, {
    name: "stigmer_workflow",
    title: "Stigmer Workflow",
    description: "Full definition of a Stigmer workflow, identified by organization and slug.",
    template: "stigmer://workflows/{org}/{slug}",
    fetch: fetchWorkflow,
  });
  return ["stigmer_workflow"];
}
